### PR TITLE
Fix a bug when getting branch commit hash from github.

### PR DESCRIPTION
**Pull request by dawelter2**

### CODE REVIEW

File: err_stash.py [hunk 1]
```diff
@@ -162,6 +162,13 @@ def fetch_repo_commits(
         return repo.compare(to_branch, from_branch).commits
 
 
+@attr.s()
```
Comment (nicoddemus) [inline err_stash.py]: Try to make it read-only. 👍

File: err_stash.py [hunk 2]
```diff
@@ -276,7 +282,7 @@ def create_plans(
 
                 plan = MergePlan(organization, repo_name, comes_from_github=True)
                 plans.append(plan)
-                plan.branches = branches
+                plan.branches = [Branch(b.name, b.name, b.commit.sha) for b in branches]
```
Comment (dawelter2) [inline err_stash.py:290]: Couldn't find the equivalent to Stash `id` in GitHub,

Comment (nicoddemus) [inline err_stash.py:290]: Put a comment here why `b.name` is being passed twice

**Review by nicoddemus — CHANGES_REQUESTED** [review]

File: err_stash.py (hunk 1 above)
Comment (dawelter2) [reply err_stash.py]: `@attr.s(frozen=True)` does the job! it is right?

File: tests.py [hunk 3]
```diff
@@ -654,67 +673,68 @@ def test_make_pr_link(github_api):
...
-    mock_stash_api["PROJ-A"]["repo1"]["pull_request"] = {"10": DummyPullRequest(True)}
-    mock_stash_api["PROJ-A"]["repo1"]["commits"] = {
-        ("refs/heads/" + from_branch, "refs/heads/target_branch"): ["A", "B"]
-    }
-
+@pytest.mark.parametrize(
+    "branch_name, expected_messages",
+    [
+        (
+            "fb-ASIM-81-network",
```
Comment (nicoddemus) [inline tests.py:684]: Please include a test: all branches on Stash

File: tests.py [hunk 4]
```diff
@@ -654,67 +673,68 @@ def test_make_pr_link(github_api):
...
-        "GitHub: deps -> (commit id *2a0ae67e039099e300ec972e22c281af19f4ac9d*) \n"
-        "*To confirm to delete this branches please _repeate_ the command*"
+    assert "\n".join(lines) == (
+        f"Found branch `{branch_name}` in these repositories:\n{expected_messages} \n*To confirm to delete this branches please _repeate_ the command*"
     )
-    assert len(branches_to_delete) == 3
+    assert len(branches_to_delete) == len(expected_messages.split("\n"))
 
 
 def test_delete_branches(mock_stash_api, github_api):
```
Comment (nicoddemus) [inline tests.py]: Please also parametrize this test so we test: deleting GH + Stash branches, only Stash branches, only GH branches. 👍

File: err_stash.py [hunk 5]
```diff
@@ -711,7 +716,7 @@ def obtain_branches_to_delete(
     yield "Found branch `{}` in these repositories:".format(branch_name)
     for plan in plans:
         branches_to_delete.append(plan)
-        yield f"{'Stash' if not plan.comes_from_github else 'GitHub'}: {plan.slug} -> (commit id *{plan.branches[0]['latestCommit']}*) {'*has PR*' if len(plan.pull_requests) > 0 else ''}"
+        yield f"{'GitHub' if plan.comes_from_github else 'Stash'}: {plan.slug} (commit id *{plan.branches[0].latest_commit}*) {'*has PR*' if len(plan.pull_requests) > 0 else ''}"
```
Comment (nicoddemus) [inline err_stash.py]: Add a property to `plan`:

```python
@property
def provider_name(self):
    return 'GitHub' if plan.comes_from_github else 'Stash'
```

And use that here:

```python
yield f"{plan.provider_name}: {plan.slug}..."
```

**Review by nicoddemus — CHANGES_REQUESTED** [review]

**Comment by nicoddemus** [comment]
I think we are nearly there. 👍

File: tests.py (hunk 3 above)
Comment (dawelter2) [reply tests.py:684]: good call!

**Review by nicoddemus — APPROVED** [review]

**Comment by nicoddemus** [comment]
great work!

**Pull request merged**